CMakeLists.txt

```diff
@@ -1113,6 +1124,7 @@ function(add_library_variant target_arch)
         SUFFIX
         COMPILE_FLAGS
         MULTILIB_FLAGS
+        BUILD_TYPE
```

Comment:
Would it be worth using `PICOLIBC_BUILD_TYPE` to make it more explicit that this is passing through --buildtype to picolibc?